File: simpleclaw_mobile/lib/models/payment.dart
class PaymentResponse {
  const PaymentResponse({
    required this.paymentId,
    required this.confirmationUrl,
    required this.yookassaId,
  });

  final int paymentId;
  final String confirmationUrl;
  final String yookassaId;

  factory PaymentResponse.fromJson(Map<String, dynamic> json) {
    return PaymentResponse(
      paymentId: json['payment_id'] as int,
      confirmationUrl: json['confirmation_url'] as String,
      yookassaId: json['yookassa_id'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'payment_id': paymentId,
      'confirmation_url': confirmationUrl,
      'yookassa_id': yookassaId,
    };
  }
}

File: simpleclaw-rn/src/types/payment.ts
export interface PaymentResponse {
  paymentId: number;
  confirmationUrl: string;
  yookassaId: string;
}

export function paymentFromJson(json: Record<string, unknown>): PaymentResponse {
  return {
    paymentId: json['payment_id'] as number,
    confirmationUrl: json['confirmation_url'] as string,
    yookassaId: json['yookassa_id'] as string,
  };
}

export interface TokenPaymentResponse {
  paymentId: number;
  yookassaId: string;
  status: string;
}

export function tokenPaymentFromJson(json: Record<string, unknown>): TokenPaymentResponse {
  return {
    paymentId: json['payment_id'] as number,
    yookassaId: json['yookassa_id'] as string,
    status: json['status'] as string,
  };
}
